plugins/kvscheduler/internal/utils/record.go

```diff
@@ -77,10 +79,17 @@ func (p *RecordedProtoMessage) UnmarshalJSON(data []byte) error {
 	if p.ProtoMsgName == "" {
 		return nil
 	}
-	msgType, err := protoregistry.GlobalTypes.FindMessageByName(protoreflect.FullName(pwn.ProtoMsgName))
+
+	typeRegistry := models.DefaultRegistry.MessageTypeRegistry()
+	fullMsgName := protoreflect.FullName(pwn.ProtoMsgName)
+	msgType, err := typeRegistry.FindMessageByName(fullMsgName)
+	if err != nil {
+		msgType, err = protoregistry.GlobalTypes.FindMessageByName(fullMsgName)
```

Comment:
Please add a comment here that this lookup in global proto registry is used as a fallback if proto model is not found in models registry.